koku/koku/celery.py

```diff
@@ -103,21 +111,66 @@ def readiness_check(self):
 
 # Toggle to enable/disable scheduled checks for new reports.
 if ENVIRONMENT.bool("SCHEDULE_REPORT_CHECKS", default=False):
+    download_expression = "0 4,16 * * *"
+    download_task = "masu.celery.tasks.check_report_updates"
     # The schedule to scan for new reports.
-    REPORT_DOWNLOAD_SCHEDULE = ENVIRONMENT.get_value(
-        "REPORT_DOWNLOAD_SCHEDULE", default="0 4,16 * * *"
-    )  # default: “At minute 0 past hour 4 and 16.”
-    if not croniter.is_valid(REPORT_DOWNLOAD_SCHEDULE):
-        print(f"Invalid report-download-schedule {REPORT_DOWNLOAD_SCHEDULE}. Falling back to default `0 4,16 * * *`")
-        REPORT_DOWNLOAD_SCHEDULE = "0 4,16 * * *"
-    report_schedule = crontab(*REPORT_DOWNLOAD_SCHEDULE.split(" ", 5))
-
-    CHECK_REPORT_UPDATES_DEF = {
-        "task": "masu.celery.tasks.check_report_updates",
-        "schedule": report_schedule,
-        "args": [],
+    REPORT_DOWNLOAD_SCHEDULE_GCP = ENVIRONMENT.get_value("REPORT_DOWNLOAD_SCHEDULE_GCP", default=download_expression)
+    REPORT_DOWNLOAD_SCHEDULE_GCP = validate_cron_expression(REPORT_DOWNLOAD_SCHEDULE_GCP)
+    report_schedule_gcp = crontab(*REPORT_DOWNLOAD_SCHEDULE_GCP.split(" ", 5))
+    CHECK_REPORT_UPDATES_DEF_GCP = {
+        "task": download_task,
+        "schedule": report_schedule_gcp,
+        "kwargs": json.dumps(
+            {
+                "provider_type": "GCP",
+            }
+        ),
+    }
+    app.conf.beat_schedule["check-report-updates"] = CHECK_REPORT_UPDATES_DEF_GCP
+
+    REPORT_DOWNLOAD_SCHEDULE_AWS = ENVIRONMENT.get_value("REPORT_DOWNLOAD_SCHEDULE_AWS", default=download_expression)
+    REPORT_DOWNLOAD_SCHEDULE_AWS = validate_cron_expression(REPORT_DOWNLOAD_SCHEDULE_AWS)
+    report_schedule_aws = crontab(*REPORT_DOWNLOAD_SCHEDULE_AWS.split(" ", 5))
+    CHECK_REPORT_UPDATES_DEF_AWS = {
+        "task": download_task,
+        "schedule": report_schedule_aws,
+        "kwargs": json.dumps(
+            {
+                "provider_type": "AWS",
+            }
+        ),
+    }
+    app.conf.beat_schedule["check-report-updates"] = CHECK_REPORT_UPDATES_DEF_AWS
+
+    REPORT_DOWNLOAD_SCHEDULE_AZURE = ENVIRONMENT.get_value(
+        "REPORT_DOWNLOAD_SCHEDULE_AZURE", default=download_expression
+    )
+    REPORT_DOWNLOAD_SCHEDULE_AZURE = validate_cron_expression(REPORT_DOWNLOAD_SCHEDULE_AZURE)
+    report_schedule_azure = crontab(*REPORT_DOWNLOAD_SCHEDULE_AZURE.split(" ", 5))
+    CHECK_REPORT_UPDATES_DEF_AZURE = {
+        "task": download_task,
+        "schedule": report_schedule_azure,
+        "kwargs": json.dumps(
+            {
+                "provider_type": "AZURE",
```

Comment:
Question:
Would this need to be
```suggestion
                "provider_type": "Azure",
```
For this check to work?
```
if provider_type and provider_type not in provider.type:
    continue
```
because of what we have set [here](https://github.com/project-koku/koku/blob/main/koku/api/provider/models.py#L61).